signature/mechanism_gpgme.go

```diff
@@ -196,6 +199,32 @@ func (m *gpgmeSigningMechanism) Verify(unverifiedSignature []byte) (contents []b
 	return signedBuffer.Bytes(), sig.Fingerprint, nil
 }
 
+// keyIdentityForVerificationKeyIdentity re-checks the key identity returned by Verify
+// if it doesn't match an identity returned by NewEphemeralGPGSigningMechanism, trying to match it.
+// (To be more specific, for mechanisms which return a subkey fingerprint from Verify,
+// this converts the subkey fingerprint into the corresponding primary key fingerprint.)
+func (m *gpgmeSigningMechanism) keyIdentityForVerificationKeyIdentity(keyIdentity string) (string, error) {
+	// In theory, if keyIdentity refers to a subkey, the same subkey could be attached to different primary keys;
+	// in that case, GetKey fails with “ambiguous name”.
+	// We _could_ handle that, by using KeyList* (GetKey is internally just a helper for KeyList*), but sharing
+	// a subkey that way is very unexpected, so, for now, prefer the much simpler implementation.
+	key, err := m.ctx.GetKey(keyIdentity, false)
+	if err != nil {
+		return "", err
+	}
+	// With https://github.com/proglottis/gpgme/pull/42, we could just
+	//	return key.Fingerprint(), nil
+	subkey := key.SubKeys()
```

Comment:
subkey returns a *SubKey, I think we'll want to iterate through this `subkey.Next()` until we hit the end. Currently the test keys only have a single signing subkey, but there is a chance that a subkey could also be an encryption or authentication subkey. Since a Primary key can have multiple signing subkeys AND signatures are only valid if they come from a signing subkey (certify key is only valid for signing another gpg key) we may need to return an slice

Reply:
I imagine a similar type of implementation will be needed in sequoia as well